Toolchain/BuildIt.sh

```diff
@@ -1,164 +0,0 @@
-        echo "Extracting binutils..."
-        tar -xzf ${BINUTILS_PKG}
-
-        pushd ${BINUTILS_NAME}
-            git init >/dev/null
-            git add . >/dev/null
-            git commit -am "BASE" >/dev/null
-            git apply "$DIR"/Patches/binutils.patch >/dev/null
-        popd
-    else
-        echo "Skipped extracting binutils"
-    fi
-
-    if [ ! -d $GCC_NAME ]; then
-        echo "Extracting gcc..."
-        tar -xzf $GCC_PKG
-
-        pushd $GCC_NAME
-            git init >/dev/null
-            git add . >/dev/null
-            git commit -am "BASE" >/dev/null
-            git apply "$DIR"/Patches/gcc.patch >/dev/null
-        popd
-    else
-        echo "Skipped extracting gcc"
-    fi
-
-    if [ "$(uname)" = "Darwin" ]; then
-        pushd "gcc-${GCC_VERSION}"
-        ./contrib/download_prerequisites
-        popd
-    fi
-
-popd
-
-mkdir -p "$PREFIX"
-
-mkdir -p "$DIR/Build/binutils"
-mkdir -p "$DIR/Build/gcc"
-
-if [ -z "$MAKEJOBS" ]; then
-    MAKEJOBS=$($NPROC)
-fi
-
-pushd "$DIR/Build/"
-    unset PKG_CONFIG_LIBDIR # Just in case
-
-    pushd binutils
-        "$DIR"/Tarballs/binutils-2.33.1/configure --prefix="$PREFIX" \
-                                                --target="$TARGET" \
-                                                --with-sysroot="$SYSROOT" \
-                                                --enable-shared \
-                                                --disable-nls || exit 1
-        if [ "$(uname)" = "Darwin" ]; then
-            # under macOS generated makefiles are not resolving the "intl"
-            # dependency properly to allow linking its own copy of
-            # libintl when building with --enable-shared.
-            "$MAKE" -j "$MAKEJOBS" || true
-            pushd intl
-            "$MAKE" all-yes
-            popd
-        fi
-        "$MAKE" -j "$MAKEJOBS" || exit 1
-        "$MAKE" install || exit 1
-    popd
-
-    pushd gcc
-        if [ `uname -s` = "OpenBSD" ]; then
-            perl -pi -e 's/-no-pie/-nopie/g' "$DIR"/Tarballs/gcc-9.2.0/gcc/configure
-        fi
-
-        "$DIR"/Tarballs/gcc-9.2.0/configure --prefix="$PREFIX" \
-                                            --target="$TARGET" \
-                                            --with-sysroot="$SYSROOT" \
-                                            --disable-nls \
-                                            --with-newlib \
-                                            --enable-shared \
-                                            --enable-languages=c,c++ || exit 1
-
-        echo "XXX build gcc and libgcc"
-        "$MAKE" -j "$MAKEJOBS" all-gcc all-target-libgcc || exit 1
-        echo "XXX install gcc and libgcc"
-        "$MAKE" install-gcc install-target-libgcc || exit 1
-
-        echo "XXX serenity libc and libm"
-        ( cd "$DIR/../Libraries/LibC/" && "$MAKE" clean && "$MAKE" && "$MAKE" install )
-        ( cd "$DIR/../Libraries/LibM/" && "$MAKE" clean && "$MAKE" && "$MAKE" install )
```

Comment:
I wonder if this was really necessary to build libstdc++-v3? Can you elaborate a bit of this use-case?
I did build LibC and LibM also with this package.sh script but I couldn't see any difference building libstdc++-v3 without doing so. Basically, in the output there was no linkage of libc/libm (.o or .a files), so my assumption is that it is not needed.
I built the toolchain without them and then run `./makeall.sh && ./sync.sh && ./run` and it all seems pretty nice (which I guess wouldn't if wrong libc/libm has been linked/used)...
If it's needed though, I will add the two scripts to build LibM and LibC in the toolchain and adapt the package.sh of gcc to add those as dependencies. Nevertheless, the current implementation here is a bit strange, because first, i686-pc-serenity-g++ is being built, then LibC and LibM is built with this new compiler and then, libstdc++-v3 is built with the host compiler again. :-D

Reply:
I'm not sure anymore. It was definitely a requirement when we were using GCC 8. Perhaps something has changed with GCC 9?

Did you try this with a fully clean build? i.e nothing in `Root/`?

Reply:
Upsi, that did the trick! I forgot to build the Root/ from scratch. Thanks for the reminder, now it isn't working anymore. I'll get on that.